core/pow/engine.go

```diff
@@ -380,6 +380,17 @@ func (e *Engine) verify(tx abci.Tx) (byte, error) {
 	// check if the transaction was seen in scope
 	txHash := hex.EncodeToString(tx.Hash())
 
+	// check for replay attacks
+	if _, ok := e.seenTx[txHash]; ok {
+		e.log.Error("replay attack: txHash already used", logging.String("tx-hash", txHash), logging.String("tid", tx.GetPoWTID()), logging.String("party", tx.Party()), logging.String("command", tx.Command().String()))
+		return h, errors.New("transaction hash already used")
+	}
+
+	// validator commands skip PoW verification
+	if tx.Command().IsValidatorCommand() {
```

Comment:
just noting here that it is a change of behaviour which means validators block is not check. I don't feel strongly against it but I don't think it's necessary either.

Reply:
Agreed, after looking at the [spec update PR](https://github.com/vegaprotocol/specs/pull/1673), I checked with @davidsiska-vega to make sure that everyone is aware of this change, and it's the intended behaviour.